PlayLoop/PlayerInteractionHandler.js

```diff
@@ -0,0 +1,201 @@
+class PlayerInteractionHandler {
+    static lastTarget = null;
+    static currentTarget = null;
+    static blockers = [];
+    static registeredTargets = [];
+    static framesOnTarget = 0;
+    static pressed = {};
+    static justPressed = {};
+    static recentlyBlocked = false;
+
+    static reset(includeControls) {
+        this.currentTarget = null;
+        this.framesOnTarget = 0;
+        this.blockers = [];
+        this.recentlyBlocked = false
+        if (includeControls) {
+            this.pressed = {};
+            this.justPressed = {};
+        }
+
+        this.clearInteractionTargets();
+    }
+
+    static clearInteractionTargets() {
+        this.registeredTargets = [];
+        this.lastTarget = this.currentTarget;
+    }
+
+    static registerInteractionTarget(interactiveObject) {
+        this.registeredTargets.push(interactiveObject);
+    }
+
+    static blockInteractionsFor(reasonString) {
+        if (!this.blockers.includes(reasonString)) {
+            this.blockers.push(reasonString);
+        }
+    }
+
+    static unblockInteractionsFor(reasonString) {
+        let index = this.blockers.indexOf(reasonString);
+        if (index > -1) {
+            this.blockers.splice(index, 1);
+        }
+    }
+
+    static isTargeting(obj) {
+        return this.currentTarget === obj;
+    }
+
+    static canTrigger(obj) {
+        return (this.isTargeting(obj) && this.blockers.length === 0);
+    }
+
+    static willAllowJump() {
+        let target = this.currentTarget || this.lastTarget;
+
+        if (!target) {
+            return true;
+        }
+
+        if (target.type === ObjectTypes.NPC) {
+            return false;
+        } 
+        if (target.type === ObjectTypes.DOOR) {
+            return (target.activationTrigger !== DOOR_TRIGGERS.UP);
+        }
+    }
+
+    static updateControl(id) {
+        // should probably merge updateControl stuff into Controller and read from there instead?
+        let buttonDown = Controller[id];
+        this.justPressed[id] = buttonDown && !this.pressed[id];
+        this.pressed[id] = buttonDown;
+    }
+
+    static handleInput() {
+        // should probably merge updateControl stuff into Controller and read from there instead?
+        // controls need to be updated every frame
+        this.updateControl('jump');
+        this.updateControl('up');
+        this.updateControl('down');
```

Comment:
if would be awesome if we would have a "just pressed" variable in controller :)
but it's ok if not

Reply:
Hah, I agree, but maybe we can add that in a future update. I think the unmerged ladder PR touched on some issues that would require a `Controller` update -- maybe it could be bundled into that?